File: core/src/main/java/ru/pocketbyte/locolaser/config/parser/EmptySourceConfigParser.java
package ru.pocketbyte.locolaser.config.parser;

import ru.pocketbyte.locolaser.config.source.EmptySourceConfig;
import ru.pocketbyte.locolaser.exception.InvalidConfigException;

public class EmptySourceConfigParser implements SourceConfigParser<EmptySourceConfig> {

    @Override
    public EmptySourceConfig parse(Object sourceObject, boolean throwIfWrongType) throws InvalidConfigException {
        if (EmptySourceConfig.TYPE.equals(sourceObject)) {
            return new EmptySourceConfig();
        }

        if (throwIfWrongType)
            throw new InvalidConfigException("Source must equal \"null\"");

        return null;
    }
}

File: core/src/main/java/ru/pocketbyte/locolaser/config/parser/EmptySourceConfigParser.kt
package ru.pocketbyte.locolaser.config.parser

import ru.pocketbyte.locolaser.config.resources.EmptyResourcesConfig
import ru.pocketbyte.locolaser.exception.InvalidConfigException
import java.io.File

@Deprecated("JSON configs is deprecated feature. You should use Gradle config configuration")
class EmptySourceConfigParser : ResourcesConfigParser<EmptyResourcesConfig> {

    @Throws(InvalidConfigException::class)
    override fun parse(resourceObject: Any?, workDir: File?, throwIfWrongType: Boolean): EmptyResourcesConfig? {
        if (EmptyResourcesConfig.TYPE == resourceObject) {
            return EmptyResourcesConfig()
        }

        if (throwIfWrongType)
            throw InvalidConfigException("Source must equal \"null\"")

        return null
    }
}
